src/primitives/cylinder.js

```diff
@@ -0,0 +1,91 @@
+const {defaultResolution3D} = require('../core/constants')
+
+const vec3 = require('../math/vec3')
+
+const geom3 = require('../geometry/geom3')
+const poly3 = require('../geometry/poly3')
+
+/** Construct a solid cylinder.
+ * @param {Object} [options] - options for construction
+ * @param {Array} [options.start=[0,-1,0]] - start point of cylinder
+ * @param {Number} [options.startRadisu=1] - radius of cylinder at the start
+ * @param {Array} [options.end=[0,1,0]] - end point of cylinder
+ * @param {Number} [options.endRadius=1] - radius of cylinder at the end
+ * @param {Number} [options.endAngle=360] - end angle of cylinder
+ * @param {Number} [options.resolution=defaultResolution3D] - number of polygons per 360 degree revolution
+ * @returns {geom3} new geometry
+ *
+ * @example
+ * let cylinder = cylinder({
+ *   start: [0, -10, 0],
```

Comment:
cylinder, cylinderElliptic, roundedCylinder use start point and end point to define the height of the cylinder. This seems counter intuitive with the other primitives which are centered, and have a Z radius. I have hit this issue many times in designs, and wonder why this was defined as such. Why not just say 'height: 10'?

also, lots of others have raised this as an issue, as the cylinder become oblique if the start and end points define a non-axis direction.

Reply:
Posted a new discussion to the forum. Let's see if there are any opinions on this.

Reply:
As a first step, cylinder is now an alias of cylinderElliptic. That should make the final changes slightly easier.

Reply:
that is great !

Reply:
cylinder() cylinderElliptic() roundedCylinder() now use 'height', removing start/end points.
i like the results.

@kaosat-dev please review again.